packages/next/src/server/config.ts

```diff
@@ -364,6 +364,30 @@ function assignDefaults(
         )
       }
 
+      // Convert URL to RemotePattern
+      images.remotePatterns = images.remotePatterns.map(
+        ({ protocol, hostname, port, pathname, search }) => {
+          if (
+            protocol &&
+            !['http', 'https', 'http:', 'https:'].includes(protocol)
+          ) {
+            throw new Error(
+              `Specified images.remotePatterns must have protocol "http" or "https" received "${protocol}".`
+            )
```

Comment:
Can we add a test for that error? Curious about the stack.

Reply:
@eps1lon Added in [754c194](https://github.com/vercel/next.js/pull/77692/commits/754c194cff571ce96b582d45b8d809f2d63d7fca)